lib/linter-view.coffee

```diff
@@ -142,48 +142,64 @@ class LinterView
   # Internal: Process the messages returned by linters and render them.
   #
   # messages - An array of messages to annotate:
-  #           :level  - the annotation error level ('error', 'warning')
+  #           :level  - the annotation error level ('error', 'warning', 'info')
   #           :range - The buffer range that the annotation should be placed
   processMessage: (messages, tempFileInfo, linter) =>
     log "#{linter.linterName} returned", linter, messages
 
+    @messages = @messages.concat(messages)
     tempFileInfo.completedLinters++
     if tempFileInfo.completedLinters == @linters.length
+      @display @messages
       rimraf tempFileInfo.path, (err) ->
         throw err if err?
 
-    @messages = @messages.concat(messages)
-    @display()
-
   # Internal: Destroy all markers (and associated decorations)
   destroyMarkers: ->
     return unless @markers?
     m.destroy() for m in @markers
     @markers = null
 
-  # Internal: Render all the linter messages
-  display: ->
+  # Internal: Create marker from message
+  createMarker: (message) ->
+    marker = @editor.markBufferRange message.range, invalidate: 'never'
+    klass = 'linter-' + message.level
+    if @showGutters
+      @editor.decorateMarker marker, type: 'gutter', class: klass
+    if @showHighlighting
+      @editor.decorateMarker marker, type: 'highlight', class: klass
+    return marker
+
+  # Internal: Pidgeonhole messages onto lines. Each line gets only one message,
+  # the message with the highest level presides.
+  sortMessagesByLine: (messages) ->
+    lines = {}
+    levels = ['info', 'warning', 'error']
+    for message in messages
+      lNum = message.line
+      line = lines[lNum] || { 'level': -1 }
+      msg_level = levels.indexOf(message.level)
+      if msg_level <= line.level
+        continue
```

Comment:
nit: `continue if...`

could also perhaps make this `continue unless msg_level > line.level`, but that's your call